hermes-consumers/src/test/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/ZookeeperWorkloadConstraintsRepositoryTest.java

```diff
@@ -0,0 +1,118 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload.constraints;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.api.TopicName;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperPaths;
+import pl.allegro.tech.hermes.test.helper.zookeeper.ZookeeperBaseTest;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static java.util.Collections.emptyMap;
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class ZookeeperWorkloadConstraintsRepositoryTest extends ZookeeperBaseTest {
+
+    private ZookeeperWorkloadConstraintsRepository repository;
+    private ZookeeperWorkloadConstraintsPathChildrenCache pathChildrenCache;
+    private ZookeeperPaths paths = new ZookeeperPaths("/hermes");
+
+    private Logger logger;
+    private ListAppender<ILoggingEvent> listAppender;
+
+    @Before
+    public void setup() {
+        logger = (Logger) LoggerFactory.getLogger(ZookeeperWorkloadConstraintsRepository.class);
+        listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
+
+        try {
+            deleteAllNodes();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        pathChildrenCache = new ZookeeperWorkloadConstraintsPathChildrenCache(zookeeperClient, paths.consumersWorkloadConstraintsPath());
+        repository = new ZookeeperWorkloadConstraintsRepository(zookeeperClient, new ObjectMapper(), paths, pathChildrenCache);
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        pathChildrenCache.close();
+    }
+
+    @Test
+    public void shouldReturnEmptyConstraintsIfBasePathDoesNotExistTest() {
```

Comment:
IMO there's no need for adding "Test" suffix to each test method :) Annotation/class names are explicit.